### PR TITLE
Change LabelSet creation to use varargs, rather than overloaded k/v methods.

### LINKED ISSUE #723: Replace labelset creation API with simple varargs creation
We can get down to a single labelset creator method if we just use varargs, rather than having all the methods with specific numbers of arguments.

### PR DESCRIPTION
**Pull request by jkwatson**
For now, keeping the version that takes a Map, and the emptyLabelSet method, although they both technically could be removed if we go with this.

resolves #723

### CODE REVIEW

File: sdk/src/test/java/io/opentelemetry/sdk/metrics/MeterSdkTest.java [hunk 1]
```diff
@@ -54,19 +53,15 @@ public void testLabelSets() {
     MeterSdk testSdk = new MeterSdk();
 
     assertThat(testSdk.emptyLabelSet()).isSameInstanceAs(testSdk.emptyLabelSet());
+    assertThat(testSdk.emptyLabelSet()).isSameInstanceAs(testSdk.createLabelSet());
     assertThat(testSdk.emptyLabelSet())
         .isSameInstanceAs(testSdk.createLabelSet(Collections.<String, String>emptyMap()));
+
     assertThat(testSdk.createLabelSet("key", "value"))
         .isEqualTo(testSdk.createLabelSet("key", "value"));
-    assertThat(testSdk.createLabelSet("k1", "v1", "k2", "v2"))
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/metrics/MeterSdkTest.java]: Why are these removed?

Comment (jkwatson) [reply sdk/src/test/java/io/opentelemetry/sdk/metrics/MeterSdkTest.java]: Hm. I guess having at least one of more than a single pair would be useful. Beyond that, it seems like the value will be very low, since varargs is varargs. I'll put the 2-key version back.

Comment (jkwatson) [reply sdk/src/test/java/io/opentelemetry/sdk/metrics/MeterSdkTest.java]: done

File: api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java [hunk 2]
```diff
@@ -125,49 +125,21 @@ public BatchRecorder newMeasureBatchRecorder() {
...
+        "You must provide an even number of key/value pair arguments.");
+    for (int i = 0; i < keyValuePairs.length; i += 2) {
+      String key = keyValuePairs[i];
+      Utils.checkNotNull(key, "You cannot provide null keys for LabelSet creation.");
+    }
     return NoopLabelSet.INSTANCE;
   }
 
   @Override
   public LabelSet createLabelSet(Map<String, String> labels) {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: Do we need this method anymore?

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: We could definitely get rid of it, if we have varargs, but I suspect there might be API usages that would want to have a static Map around to use. I'd like to get some feedback on the varargs option before we remove the Map option. We could also get rid of the empty method, since you can just call the varargs one with no arguments to get the empty LabelSet back.

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: Definitely the empty api does not make that much sense. I am for removing the empty and maybe ask some users for feedback for the map?

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: Sounds good to me. I'll make those changes and update the PR.

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: Do you want to do the changes in this PR or separately?

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java]: updated in this PR already. :)

File: api/src/test/java/io/opentelemetry/OpenTelemetryTest.java [hunk 3]
```diff
@@ -371,27 +371,7 @@ public BatchRecorder newMeasureBatchRecorder() {
...
-        String k1, String v1, String k2, String v2, String k3, String v3) {
-      return null;
-    }
-
-    @Nullable
-    @Override
-    public LabelSet createLabelSet(
-        String k1, String v1, String k2, String v2, String k3, String v3, String k4, String v4) {
+    public LabelSet createLabelSet(String... keyValuePairs) {
       return null;
```
Comment (bogdandrutu) [inline api/src/test/java/io/opentelemetry/OpenTelemetryTest.java]: We should return a NoopLabelSet to not crash if they call record on a metric with the result of this.

Comment (jkwatson) [reply api/src/test/java/io/opentelemetry/OpenTelemetryTest.java]: this is in test code...
